libs/config/yamlloader/loader.go

```diff
@@ -0,0 +1,227 @@
+package yamlloader
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/databricks/cli/libs/config"
+	"gopkg.in/yaml.v3"
+)
+
+type loader struct {
+	path string
+}
+
+func errorf(loc config.Location, format string, args ...interface{}) error {
+	return fmt.Errorf("yaml (%s): %s", loc, fmt.Sprintf(format, args...))
+}
+
+func newLoader(path string) *loader {
+	return &loader{
+		path: path,
+	}
+}
+
+func (d *loader) location(node *yaml.Node) config.Location {
+	return config.Location{
+		File:   d.path,
+		Line:   node.Line,
+		Column: node.Column,
+	}
+}
+
+func (d *loader) load(node *yaml.Node) (config.Value, error) {
+	loc := config.Location{
+		File:   d.path,
+		Line:   node.Line,
+		Column: node.Column,
+	}
+
+	var value config.Value
+	var err error
+
+	switch node.Kind {
+	case yaml.DocumentNode:
+		value, err = d.loadDocument(node, loc)
+	case yaml.SequenceNode:
+		value, err = d.loadSequence(node, loc)
+	case yaml.MappingNode:
+		value, err = d.loadMapping(node, loc)
+	case yaml.ScalarNode:
+		value, err = d.loadScalar(node, loc)
+	case yaml.AliasNode:
+		value, err = d.loadAlias(node, loc)
+	default:
+		return config.NilValue, errorf(loc, "unknown node kind: %v", node.Kind)
+	}
+
+	if err != nil {
+		return value, err
+	}
+
+	// Mark value as anchor if needed.
+	// If this node doesn't map to a type, we don't need to warn about it.
+	if node.Anchor != "" {
+		value = value.MarkAnchor()
+	}
+
+	return value, nil
+}
+
+func (d *loader) loadDocument(node *yaml.Node, loc config.Location) (config.Value, error) {
+	return d.load(node.Content[0])
+}
+
+func (d *loader) loadSequence(node *yaml.Node, loc config.Location) (config.Value, error) {
+	acc := make([]config.Value, len(node.Content))
+	for i, n := range node.Content {
+		v, err := d.load(n)
+		if err != nil {
+			return config.NilValue, err
+		}
+
+		acc[i] = v
+	}
+
+	return config.NewValue(acc, loc), nil
+}
+
+func (d *loader) loadMapping(node *yaml.Node, loc config.Location) (config.Value, error) {
+	var merge *yaml.Node
+
+	acc := make(map[string]config.Value)
+	for i := 0; i < len(node.Content); i += 2 {
+		key := node.Content[i]
+		val := node.Content[i+1]
+
+		// Assert that keys are strings
+		if key.Kind != yaml.ScalarNode {
+			return config.NilValue, errorf(loc, "key is not a scalar")
+		}
+
+		st := key.ShortTag()
+		switch st {
+		case "!!str":
+			// OK
+		case "!!merge":
+			merge = val
+			continue
```

Comment:
I guess there is only one merge node in each mapping? Any sense in panicking if merge is non-nil and we hit this codepath?